ambassador/ambassador/config.py

```diff
@@ -1096,6 +1096,14 @@ def generate_intermediate_config(self):
             # ...and route.
             self.add_intermediate_route(mapping['_source'], mapping, svc, cluster_name)
 
+            # If x_forwarded_proto_redirect is set, then we enable require_tls in primary listener, which in turn sets
+            # require_ssl to true in envoy config. Once set, then all requests that contain X-FORWARDED-PROTO set to
+            # https, are processes normally by envoy. In all the other cases, including X-FORWARDED-PROTO set to http,
+            # a 301 redirect response to https://host is sent
+            if mapping.get('x_forwarded_proto_redirect') is not None:
```

Comment:
Let's require a True value here, because it'll be very surprising if someone tries to disable it by setting `x_forwarded_proto_redirect: false` and it stays on. :)

Reply:
+1, will also add a test to catch that

Reply:
fixed